react_app/src/components/Misc/PlayerDetails.js

```diff
@@ -0,0 +1,39 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+
+import Button from "@material-ui/core/Button";
+
+class PlayerDetails extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  getUserName() {
+    return this.props.userName;
```

Comment:
Extracting things like this out is something that you will eventually regret! There are loads of ways to cut down on code - one of them is to stop using classes!!! especially in situations like this where youv'e got no state or life cycle going on. 

Destructuring is another way. Here's what your `render` method could look like;

```
  render() {
    const { isAuthenticated, userName, teamName } = this.props;

    if (!isAuthenticated)
      return null;
    
    return (
      <Button color="inherit">
        {userName} - [{teamName || "Solo" }]
      </Button>
    );
  }
}
```

Reply:
Yeah, this is better javascript, although this doesn't seem as readable as my more verbose attempt